prow/github/hmac.go

```diff
@@ -46,3 +64,37 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// For given repository/organization, returns all *valid* HMAC tokens. Validity is determined based on time till they are valid.
+func ExtractHmacs(repo string, tokenGenerator func() []byte) [][]byte {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]map[string]time.Time{}
+
+	err := yaml.Unmarshal(t, &repoToTokenMap)
+	if err != nil {
+		fmt.Printf("error: %v", err)
+		return nil
+	}
+
+	orgName := strings.Split(repo, "/")[0]
+	var extractValidToken = func(allTokens map[string]time.Time) [][]byte {
+		var validTokens [][]byte
+		for key, value := range allTokens {
+			if value.After(time.Now()) {
+				validTokens = append(validTokens, []byte(key))
+			}
+		}
+		return validTokens
+	}
+
+	if val, ok := repoToTokenMap[repo]; ok {
+		return extractValidToken(val)
+	}
+	if val, ok := repoToTokenMap[orgName]; ok {
+		return extractValidToken(val)
+	}
+	if val, ok := repoToTokenMap["global_hmac"]; ok {
```

Comment:
While I admit it is *more* descriptive, I am curious why we chose `global_hmac` over `*` here ([ref](https://docs.google.com/document/d/1m3nFfPt95kbe6BQdWp5BrZMSPkBo5mFb2nuMq1OeE24/edit?disco=AAAAI7hQyF8)); the latter is what we *consistently* use elsewhere in Prow configuration to denote globals/wildcards (e.g. `job_url_prefix_config`, `default_decoration_configs`, `slack_reporter_configs`, etc)?

Reply:
`*` would be a more consistent choice, +1

Reply:
Done.